src/components/Accordion/Accordion.tsx

```diff
@@ -62,16 +62,44 @@ export const AccordionItem = ({
   )
 }
 
+function buildExpansions(
+  items: AccordionItemProps[],
+  multiselectable: boolean,
+  savedExpansions = new Map<string, boolean | undefined>()
+) {
+  const lastExpandedItem = items.findLast(
+    (item) => item.expanded || savedExpansions.get(item.id)
+  )
+  return items.reduce((map, item) => {
+    map.set(
+      item.id,
+      !multiselectable
+        ? !!lastExpandedItem && item.id === lastExpandedItem.id
+        : (savedExpansions.get(item.id) ?? !!item.expanded)
```

Comment:
nit: I believe we can simplify this thanks to Typescript (even though there was a similar simplification possible in the code before this PR)

```suggestion
        : (savedExpansions.get(item.id) ?? item.expanded)
```